docs/source/hooks.rst

```diff
@@ -0,0 +1,158 @@
+.. _hooks:
+
+=====
+Hooks
+=====
+
+.. _hooks_introduction:
+
+Introduction
+============
+
+One of the goals of the **signac** framework is to make it easy to track the provenance of research data and to ensure its reproducibility.
+Hooks make it possible to track state changes to each job in a **signac** project as the :ref:`FlowProject<flow-project>` operates on it.
+
+A hook is a function that is called at a specific point relative to the execution of a **signac-flow** :ref:`operation<operations>`.
+A hook is triggered when an operation starts, exits, succeeds, or fails.
+
+A basic use case is to log the success/failure of an operation by creating a hook that sets a job document value ``job.doc.operation_success`` to ``True`` or ``False``.
+Additionally, a user may record the `git commit ID <https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History>`_ upon the start of an operation, allowing them to track which version of code ran the operation.
+
+.. _hook_triggers:
+
+Hook Triggers
+=============
+
+The following triggers are provided:
+
+1. :py:meth:`~flow.FlowProject.add_hook.on_start` will execute when the operation begins execution.
+2. :py:meth:`~flow.FlowProject.add_hook.on_exit` will execute when the operation exits, with or without error.
+3. :py:meth:`~flow.FlowProject.add_hook.on_success` will execute when the operation exits without error.
+4. :py:meth:`~flow.FlowProject.add_hook.on_error` will execute when the operation exits with error.
+
+Hooks can be installed at the :ref:`operation level <operation hooks>` or at the :ref:`flow-project level<project-level hooks>`.
+Project-level hooks are called for every operation in the flow project.
+
+.. note::
+
+    Hooks are run in the environment of the python process from which you call FlowProject.main().
```

Comment:
Need to add the proper reference to how flow is run.

Reply:
```suggestion
    Hooks are run in the Python process where ``FlowProject.main()`` is called.
```